internal/services/streamanalytics/stream_analytics_output_cosmosdb_resource_test.go

```diff
@@ -0,0 +1,241 @@
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cosmosdb_account_key"),
+	})
+}
+
+func TestAccStreamAnalyticsOutputCosmosDB_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stream_analytics_output_cosmosdb", "test")
+	r := StreamAnalyticsOutputCosmosDBResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("cosmosdb_account_key"),
+	})
+}
+
+func TestAccStreamAnalyticsOutputCosmosDB_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_stream_analytics_output_cosmosdb", "test")
+	r := StreamAnalyticsOutputCosmosDBResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+func (r StreamAnalyticsOutputCosmosDBResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.OutputID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.StreamAnalytics.OutputsClient.Get(ctx, id.ResourceGroup, id.StreamingjobName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r StreamAnalyticsOutputCosmosDBResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_stream_analytics_output_cosmosdb" "test" {
+  name                     = "acctestoutput-%[3]d"
+  stream_analytics_job_id  = azurerm_stream_analytics_job.test.id
+  cosmosdb_account_key     = azurerm_cosmosdb_account.test.primary_key
+  cosmosdb_sql_database_id = azurerm_cosmosdb_sql_database.test.id
+  collection_name_pattern  = azurerm_cosmosdb_sql_container.test.id
+}
+`, template, data.RandomString, data.RandomInteger)
+}
+
+func (r StreamAnalyticsOutputCosmosDBResource) updated(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_cosmosdb_sql_database" "updated" {
+  name                = "updated-cosmos-sql-db"
+  resource_group_name = azurerm_cosmosdb_account.test.resource_group_name
+  account_name        = azurerm_cosmosdb_account.test.name
+  throughput          = 400
+}
+
+resource "azurerm_cosmosdb_sql_container" "updated" {
+  name                = "updated-container%[2]s"
+  resource_group_name = azurerm_cosmosdb_account.test.resource_group_name
+  account_name        = azurerm_cosmosdb_account.test.name
+  database_name       = azurerm_cosmosdb_sql_database.updated.name
+  partition_key_path  = "/definition/id"
```

Comment:
This is a nested partition key which isn't supported by Stream Analytics, an error will be thrown when testing the connectivity
```suggestion
  partition_key_path  = "/definition"
```